gps/cmd.go

```diff
@@ -4,6 +4,49 @@
 
 package gps
 
+import (
+	"context"
+	"fmt"
+	"os/exec"
+)
+
+type ctxKey int
+
+const subProcsSem ctxKey = 0
+
+type sem chan struct{}
+
+// CtxWithCmdLimit returns a copy ctx with a semaphore value limiting the number
+// of concurrent `cmd`s. Returns an error if n is not positive.
+func CtxWithCmdLimit(ctx context.Context, n int) (context.Context, error) {
```

Comment:
This is available for anyone using package `gps`, `dep`, or `cmd/dep` as a library.